apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/AssessmentQuestionManualGrading.tsx

```diff
@@ -0,0 +1,205 @@
+        requestBody.closed_instance_questions_only = closedOnly;
+      } else {
+        requestBody.closed_instance_questions_only = false;
+      }
+
+      const response = await fetch('', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestBody),
+      });
+
+      if (!response.ok) {
+        throw new Error('Request failed with status ' + response.status);
+      }
+
+      const data = await response.json();
+      return data as { jobSequenceId?: string; success: boolean };
+    },
+    onSuccess: (data) => {
+      if (data.jobSequenceId) {
+        window.location.href = `${urlPrefix}/jobSequence/${data.jobSequenceId}`;
+      }
+    },
+    onError: (error) => {
+      console.error('Group submission failed:', error);
+    },
+  });
+
+  return (
+    <>
+      <AssessmentQuestionTable
+        authzData={authzData}
+        course={course}
+        courseInstance={courseInstance}
+        csrfToken={csrfToken}
+        instanceQuestions={instanceQuestions}
+        urlPrefix={urlPrefix}
+        assessmentId={assessmentId}
+        assessmentQuestion={assessmentQuestion}
+        assessmentTid={assessmentTid}
+        questionQid={questionQid}
+        aiGradingMode={aiGradingMode}
+        groupWork={groupWork}
+        rubricData={rubricData}
+        instanceQuestionGroups={instanceQuestionGroups}
+        courseStaff={courseStaff}
+        aiGradingStats={aiGradingStats}
+        onShowGroupSelectedModal={handleShowSelectedModal}
+        onShowGroupAllModal={() => setShowAllModal(true)}
+        onShowGroupUngroupedModal={() => setShowUngroupedModal(true)}
+      />
+
+      <GroupInfoModal
+        modalFor="selected"
+        numOpenInstances={numOpenInstances}
+        show={showSelectedModal}
+        onHide={() => setShowSelectedModal(false)}
+        onSubmit={(closedOnly) =>
+          groupSubmissionMutation.mutate({
+            action: 'batch_action',
+            closedOnly,
+            instanceQuestionIds: selectedIdsForGrouping,
+          })
+        }
+      />
+
+      <GroupInfoModal
+        modalFor="all"
+        numOpenInstances={numOpenInstances}
+        show={showAllModal}
+        onHide={() => setShowAllModal(false)}
+        onSubmit={(closedOnly) =>
+          groupSubmissionMutation.mutate({
+            action: 'ai_instance_question_group_assessment_all',
+            closedOnly,
+          })
+        }
+      />
+
+      <GroupInfoModal
+        modalFor="ungrouped"
+        numOpenInstances={numOpenInstances}
+        show={showUngroupedModal}
+        onHide={() => setShowUngroupedModal(false)}
+        onSubmit={(closedOnly) =>
+          groupSubmissionMutation.mutate({
+            action: 'ai_instance_question_group_assessment_ungrouped',
+            closedOnly,
+          })
+        }
+      />
+    </>
+  );
+}
+
+export function AssessmentQuestionManualGrading({
```

Comment:
This wrapper is needed as we need mutations in a component wrapped by `QueryClientProviderDebug`.